nva-datamodel-java/src/main/java/no/unit/nva/model/associatedartifacts/file/PublisherVersion.java

```diff
@@ -0,0 +1,57 @@
+package no.unit.nva.model.associatedartifacts.file;
+
+import static java.lang.String.format;
+import static java.util.Arrays.stream;
+import static java.util.Objects.isNull;
+import static java.util.stream.Collectors.joining;
+import com.fasterxml.jackson.annotation.JsonValue;
+import no.unit.nva.model.Revision;
+
+public enum PublisherVersion {
+    PUBLISHED_VERSION("PublishedVersion"),
+    ACCEPTED_VERSION("AcceptedVersion");
+
+    public static final String ERROR_MESSAGE_TEMPLATE = "%s not a valid PublisherVersion, expected one of: %s";
+    public static final String ERROR_MESSAGE_CANNOT_PARSE_THIS_OBJECT = "%s not a valid object, expected either "
+                                                                        + "String or boolean";
+    public static final String DELIMITER = ", ";
+
+    private final String value;
+
+    PublisherVersion(String value) {
+        this.value = value;
+    }
+
+    public static PublisherVersion parse(Object candidate) {
```

Comment:
The intention here is to make it backwards compatible until we have performed a batch scan.

Reply:
Can it be a fancy switch, or have we not bumped the java? I guess not.

Reply:
java 17 support those new types of switch, but the switch can't handle Object type of reference it seems:

![image](https://github.com/BIBSYSDEV/nva-datamodel-java/assets/44363838/953a37ed-9c01-4cbc-87f9-021730d5e80a)

Reply:
and isNull check inside the case wasn't allowed either.